### PR TITLE
Add webhook presets

### PR DESCRIPTION
**Pull request by mderynck**
# What this PR does
Add a system similar to how we select integrations when creating webhooks so that the user has a description of what webhookds do and does not have to write complex templates for common webhook use cases. Presets allow us to create the contents of the webhooks in code and define which fields are controlled by the preset. Some specifics:

- Newly created webhooks must choose between Simple, Advanced or another predefined system
- Simple is always an escalation step and will post the entire payload to the given URL
- Advanced is the same as no preset which is our current view where all fields are available
- There are no changes for all existing webhooks with empty preset fields
- Once a webhook is created with a preset the preset cannot be changed
- Fields in the webhook that are populated by code will give a validation error if they are modified
- In the public API webhooks with presets are returned for viewing but cannot be created or modified. This restriction is in place because the Web UI provides the context for which fields to use with a preset.  The public API is for interacting with webhooks where all fields are defined.

To define a preset create a file with metadata and an override function.  The metadata drives validation and what to display in the UI.  There are two functions one is connected to the pre_save hook of the Webhook model for persistent changes, the other replaces parameters at execution time for ephemeral changes.  See the simple and advanced presets as an example.  The file must be listed in settings in `INSTALLED_WEBHOOK_PRESETS` to be enabled at runtime..  

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

**Review by matiasb — APPROVED** [review]
Backend changes look good to me.

File: grafana-plugin/src/components/IntegrationLogo/IntegrationLogo.module.css [hunk 1]
```diff
@@ -24,6 +24,12 @@
   background-size: 100% !important;
 }
 
+.bg_GrafanaSift,
+.bg_GrafanaMachineLearning {
+  background: url(../../assets/img/grafana_ml_icon.svg);
+  background-size: 100% !important;
+}
+
```
[the remark this answers is not in the document]
Comment (mderynck) [reply grafana-plugin/src/components/IntegrationLogo/IntegrationLogo.module.css:32]: Removed, we will sort it out in the other PR

File: engine/apps/api/serializers/webhook.py [hunk 2]
```diff
@@ -59,11 +60,8 @@ class Meta:
             "trigger_type_name",
             "last_response_log",
             "integration_filter",
+            "preset",
         ]
-        extra_kwargs = {
-            "name": {"required": True, "allow_null": False, "allow_blank": False},
```
[the remark this answers is not in the document]
Comment (mderynck) [reply engine/apps/api/serializers/webhook.py:64]: Yes `test_create_invalid_missing_fields` is passing which checks this condition.  `UniqueTogetherValidator` forces required on name and organization.

File: docs/sources/outgoing-webhooks/_index.md [hunk 3]
```diff
@@ -30,8 +30,10 @@ Jinja2 templates to customize the request being sent.
 ## Creating an outgoing webhook
 
 To create an outgoing webhook navigate to **Outgoing Webhooks** and click **+ Create**. On this screen outgoing
-webhooks can be viewed, edited and deleted. To create the outgoing webhook populate the required fields and
-click **Create Webhook**
+webhooks can be viewed, edited and deleted. To create the outgoing webhook click **New Outgoing Webhook** and then
+select a preset based on what you want to do. A simple webhook will POST alert group data as a selectable escalation
+step to the specified url. If you require more customization use the advanced webhook which provides all of the
+field described below.
```
Comment (joeyorlando) [inline docs/sources/outgoing-webhooks/_index.md:36]:
```suggestion
fields described below.  
```

**Review by joeyorlando — APPROVED** [review]
backend changes LGTM. I really like the backend design for presets 👍

**Review by teodosii — APPROVED** [review]

**Pull request merged**